src/control/lib/hardware/topology.go

```diff
@@ -182,6 +244,62 @@ func (t *Topology) Merge(newTopo *Topology) error {
 			}
 		}
 	}
+	return t.mergeVirtualDevices(newTopo)
+}
+
+func (t *Topology) mergeVirtualDevices(newTopo *Topology) error {
+	curDevices := t.AllDevices()
+
+	for _, newVirt := range newTopo.VirtualDevices {
+		if curDev, found := curDevices[newVirt.Name]; found {
+			curVirt, ok := curDev.(*VirtualDevice)
+			if !ok {
+				return errors.Errorf("virtual device %q has same name as a hardware device", newVirt.Name)
+			}
+
+			if newVirt.Type != DeviceTypeUnknown {
+				curVirt.Type = newVirt.Type
```

Comment:
Should we at least log a warning if curVirt.Type is different from DeviceTypeUnknown or is it OK to lost this information ?

Reply:
In theory I wouldn't really expect it to happen, though maybe that's a good reason to log it. I could address this in a follow-on PR.

Reply:
OK